src/librustc_trans/mir/block.rs

```diff
@@ -95,17 +96,31 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
                     adt::trans_get_discr(bcx, &repr, discr_lvalue.llval, None, true)
                 );
 
-                // The else branch of the Switch can't be hit, so branch to an unreachable
-                // instruction so LLVM knows that
-                let unreachable_blk = self.unreachable_block();
-                let switch = bcx.switch(discr, unreachable_blk.llbb, targets.len());
+                let mut bb_hist = FnvHashMap();
```

Comment:
There's also the `assert_eq!(adt_def.variants.len(), targets.len());` which should in theory make sure we've covered all the cases.